test/test_field.jl

```diff
@@ -451,4 +450,54 @@ end
             end
         end
     end
+
+    @testset "Views of field views" begin
+        @info "  Testing views of field views..."
+
+        Nx, Ny, Nz = 1, 1, 7
+
+        FieldTypes = (CenterField, XFaceField, YFaceField, ZFaceField)
+        ZTopologies = (Periodic, Bounded)
+
+        for arch in archs, FT in float_types, FieldType in FieldTypes, ZTopology in ZTopologies
+            grid = RectilinearGrid(arch, FT, size=(Nx, Ny, Nz), x=(0, 1), y=(0, 1), z=(0, 1), topology = (Periodic, Periodic, ZTopology))
+            Hx, Hy, Hz = halo_size(grid)
+
+            k_top = (FieldType == ZFaceField && ZTopology == Bounded) ? Nz+1 : Nz 
+
+            c = FieldType(grid)
+            set!(c, (x, y, z) -> rand())
+
+            # First test that the regular view is correct
+            cv = view(c, :, :, 1+1:k_top-1)
+            @test size(cv) == (1, 1, k_top-2)
+            @test size(parent(cv)) == (1+2Hx, 1+2Hy, k_top-2)
+            CUDA.@allowscalar @test all(cv[1, 1, i] == c[1, 1, i] for i in 1+1:k_top-1)

```

Comment:
```suggestion
            CUDA.@allowscalar @test all(cv[1, 1, i] == c[1, 1, i] for i in 1+1:k_top-1)

```

Reply:
@siddharthabishnu could you remove all the rest hardcoded 1's below (unless they are not hard coded and I missed it...)